oli/src/commands/profile.rs

```diff
@@ -0,0 +1,148 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+use std::env;
+use std::str::FromStr;
+
+use anyhow::anyhow;
+use anyhow::Result;
+
+use opendal::Operator;
+
+const OLI_PROFILE_PREFIX: &str = "OLI_PROFILE_";
+
+pub struct OliProfileGroup {
+    pub name: String,
+    pub typ: String,
+    pub values: HashMap<String, String>,
+}
+
+impl OliProfileGroup {
+    pub fn parse(name: &str, kvs: HashMap<String, String>) -> Result<Self> {
+        if let Some(typ) = kvs.get("type") {
+            Ok(OliProfileGroup {
+                name: name.to_string(),
+                typ: typ.to_string(),
+                values: kvs,
+            })
+        } else {
+            Err(anyhow!("oli profile must contains type"))
+        }
+    }
+}
+
+pub(crate) fn build_operators() -> Result<HashMap<String, Operator>> {
```

Comment:
We don't need to build all operators at once. We can store them and build them on demand.

Reply:
Ok, change to lazy load.